exporter/awsemfexporter/metric_translator.go

```diff
@@ -118,6 +134,278 @@ func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
+			dps = DoubleDataPointSlice{metric.DoubleGauge().DataPoints()}
+		case pdata.MetricDataTypeIntSum:
+			dps = IntDataPointSlice{metric.IntSum().DataPoints()}
+		case pdata.MetricDataTypeDoubleSum:
+			dps = DoubleDataPointSlice{metric.DoubleSum().DataPoints()}
+		case pdata.MetricDataTypeDoubleHistogram:
+			dps = DoubleHistogramDataPointSlice{metric.DoubleHistogram().DataPoints()}
+		default:
+			config.logger.Warn("Unhandled metric data type.",
+				zap.String("DataType", metric.DataType().String()),
+				zap.String("Name", metric.Name()),
+				zap.String("Unit", metric.Unit()),)
+			return
+	}
+
+	if dps.Len() == 0 {
+		return
+	}
+
+	for m := 0; m < dps.Len(); m++ {
+		dp := dps.At(m)
+		if dp.IsNil() {
+			continue
+		}
+
+		fields := make(map[string]interface{})
+		s := []string{nameSpace, namespace, OTellibDimensionKey, instrumentationLibName}
+		fields[nameSpace] = namespace
+		fields[OTellibDimensionKey] = instrumentationLibName
+
+		dp.LabelsMap().ForEach(func(k string, v string) {
+			fields[k] = v
+			if (v != noInstrumentationLibraryName) {
+				s =append(s, k, v)
+			}
+		})
+
+		sort.Strings(s)
+		key := strings.Join(s, "")
+
+		if _, ok := groupedMetricMap[key]; ok {
+			updateGroupedMetric(dp, metric, key, groupedMetricMap)
+			groupedMetrics = append(groupedMetrics, groupedMetricMap[key])
+		} else {
+			groupedMetric := buildGroupedMetric(dp, fields, metric, config.DimensionRollupOption)
+			if groupedMetric != nil {
+				groupedMetricMap[key] = groupedMetric
+				groupedMetrics = append(groupedMetrics,groupedMetric)
+			}
+		}
+	}
+	return
+}
+
+// buildGroupedMetric builds GroupedMetric from Datapoint and pdata.Metric
+func buildGroupedMetric (dp DataPoint, fields map[string]interface{}, pMetricData *pdata.Metric, dimensionRollupOption string) (*GroupedMetric) {
+	var namespace string
+	timestamp := time.Now().UnixNano() / int64(time.Millisecond)
+	labels := make(map[string]string)
+	metrics := make(map[string]MetricInfo)
+	
+	namespace = fields[nameSpace].(string)
+
+	// Extract labels
+	for k, v := range fields {
+		if k != nameSpace {
+			labels[k] = v.(string)
+		}
+	}
+
+	// Extract metric
+	var metricVal interface{}
+	switch metric := dp.(type) {
+	case pdata.IntDataPoint:
+		metricVal = int64(metric.Value())
+		if needsCalculateRate(pMetricData) {
+			metricVal = calculateRate(fields, metric.Value(), timestamp)
+		}
+	case pdata.DoubleDataPoint:
+		metricVal = float64(metric.Value())
+		if needsCalculateRate(pMetricData) {
+			metricVal = calculateRate(fields, metric.Value(), timestamp)
+		}
+	case pdata.DoubleHistogramDataPoint:
+		bucketBounds := metric.ExplicitBounds()
+		metricVal = &CWMetricStats{
+			Min:   bucketBounds[0],
+			Max:   bucketBounds[len(bucketBounds)-1],
+			Count: metric.Count(),
+			Sum:   metric.Sum(),
+		}
+	}
+	if metricVal == nil {
+		return nil
+	}
+	metricInfo := &MetricInfo {
+		Value: metricVal,
+		Unit:  pMetricData.Unit(),
+	}
+
+	metrics[pMetricData.Name()] = *metricInfo
```

Comment:
Why do we dereference it here? Should it be a pointer (refer to comment above: https://github.com/hdj630/opentelemetry-collector-contrib/pull/95/files#r524629096)?